### DISCUSSION TITLE
Variant composition roadmap?

### DISCUSSION THREAD

**Discussion by parkersweb** [system-ui/theme-ui#1526] [Q&A]
Hey all

We're migrating an existing component library to styled-system at the moment - and would love to move to theme-ui for better typescript support. Lack of variant composition is preventing us from doing that however as it's required for some components.

 #832 indicates it's in the v1 roadmap - but the PR is looking relatively close to merge. Is it possible it'll arrive in 0.5 or 0.6? We're about to undertake a relatively complex migration process of the current library - so would rather do this in one pass if at all possible 😄 

Thanks!

Andrew

**Comment by lachlanjc** [comment]
The 3 of us core maintainers all work on Theme UI unpaid outside of our full-time jobs, so though we make new alpha releases every few weeks, we can't give an exact timeline for new features. That said, if you'd be willing to contribute, I think that PR is pretty close, & variant composition has personally been my most-wanted feature for 2 years, so we'd love any assistance. I'm glad you're liking Theme UI, & let us know how things go!

**Reply by parkersweb** [reply]
Thanks so much @lachlanjc - my experience on building libraries like this is fairly limited but I'll definitely have a look to see if I can help contribute (although PR #1226 looks like it's not far off?). Thanks for all you guys are doing!
